packages/core/src/browser/quick-input/quick-command-service.ts

```diff
@@ -0,0 +1,156 @@
+/********************************************************************************
+ * Copyright (c) 2021 SAP SE or an SAP affiliate company and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { inject, injectable } from 'inversify';
+import { Disposable, Command, CommandRegistry } from '../../common';
+import { ContextKeyService } from '../context-key-service';
+import { CorePreferences } from '../core-preferences';
+import { QuickAccessContribution } from './quick-access-contribution';
+import { QuickInputService } from './quick-input-service';
+
+export const quickCommand: Command = {
+    id: 'workbench.action.showCommands'
+};
+
+export const CLEAR_COMMAND_HISTORY: Command = {
+    id: 'clear.command.history',
+    label: 'Clear Command History'
+};
+
+@injectable()
+export class QuickCommandService extends QuickInputService implements QuickAccessContribution {
+
+    @inject(ContextKeyService)
+    protected readonly contextKeyService: ContextKeyService;
+
+    @inject(CommandRegistry)
+    protected readonly commandRegistry: CommandRegistry;
+
+    @inject(CorePreferences)
+    protected readonly corePreferences: CorePreferences;
+
+    // The list of exempted commands not to be displayed in the recently used list.
+    readonly exemptedCommands: Command[] = [
+        CLEAR_COMMAND_HISTORY,
+    ];
+
+    registerQuickAccessProvider(): void { }
+
+    protected readonly contexts = new Map<string, string[]>();
+    pushCommandContext(commandId: string, when: string): Disposable {
+        const contexts = this.contexts.get(commandId) || [];
+        contexts.push(when);
+        this.contexts.set(commandId, contexts);
+        return Disposable.create(() => {
+            const index = contexts.indexOf(when);
```

Comment:
consider `includes`

Reply:
1. Not sure how to use `includes`. Code is using index to remove item
``` 
const index = contexts.indexOf(when);
if (index !== -1) {
    contexts.splice(index, 1);
}
```

2. This code existed originally in previous version - I didn't change it.